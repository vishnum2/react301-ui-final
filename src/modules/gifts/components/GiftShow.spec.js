import React from 'react';
import Giftshow from './GiftShow';
import { shallow, mount } from '../../../enzyme';

describe('snapshot of Giftshow', () => {
    it('captures and tests snapshot', () => {
        const wrapper = shallow(<Giftshow data={{}}/>);
        expect(wrapper).toMatchSnapshot();
    })
});

describe('getEmail function mock', () => {
    it('should mock function and run', () => {
        const props =  jest.fn();
        const wrapper = shallow(<Giftshow data={{cardComments: [{comment: 'erf', first_name: '', last_name: '', commented_On: '', rating: ''}]}} getEmail={props}/>);
        wrapper.find('DraggableDialog').props().getEmail();
        expect(props).toHaveBeenCalled();
        
    })
})

describe('sendCardDialog renders in GiftShow', () => {
    it('should render sendcardDialog', () => {
        const wrapper = mount(<Giftshow data={{}} />);
        // console.log(wrapper.find({testData: 'sendGiftCardDialog'}).debug());
        expect(wrapper.find({testData: 'sendGiftCardDialog'})).toHaveLength(1);
    })
})