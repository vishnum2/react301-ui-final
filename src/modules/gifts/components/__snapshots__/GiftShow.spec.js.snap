// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of Giftshow captures and tests snapshot 1`] = `ShallowWrapper {}`;
